### PR TITLE
VP-1230 Delete and Edit a Blog Post

### PR DESCRIPTION
**Pull request by farjanacan**
## I do solemnly swear that I have:
- [x] Run `npm test` and all the tests pass.
- [x] Run `npm run lint` and there are no warnings.
- [x] Not decreased the overall test coverage?
- [x] Included the Jira ID and description in the PR title

## Proposed Changes? 🤔
1. Blog posts can be edited
2. Blog posts can be removed
3. Blog posts are now sorted by date added
4. Blog posts exceeding max length can be expanded with a 'read more' option

### Original
original

### Updated
updated

### CODE REVIEW

File: server/api/story/__tests__/story.spec.js [hunk 1]
```diff
@@ -24,45 +26,82 @@ test.serial('verify fixture database has stories', async t => {
...
-    .get('/api/stories')
+    .get('/api/stories/?parentId=5df0265916dc014f404ce0a0')
     .set('Accept', 'application/json')
-    .set('Cookie', [`idToken=${jwtData.idToken}`])
     .expect(200)
     .expect('Content-Type', /json/)
   const got = res.body
-  t.is(2, got.length)
-  t.is(got[0].name, 'Voluntarily wins Open Source Web prize')
+  t.is(1, got.length)
```
Comment (avowkind) [inline server/api/story/__tests__/story.spec.js]: good to check you get 1 result - but you also need to check it is the result you are expecting and not just the first one in the list. e.g t.is(got.parent,  parentId). probably best to add two stories with the same parent to the fixture.

File: components/Story/AddStory.js [hunk 2]
```diff
@@ -9,14 +9,16 @@ import { connect } from 'react-redux'
...
   const [showForm, setShowForm] = useState(false)
-  const handleSave = story => {
+
+  const handleSave = (story) => {
     setShowForm(false)
     onSubmit(story)
   }
-  if (roles && (roles.includes(Role.ADMIN) || roles.includes(Role.ORG_ADMIN))) {
+
+  if (roles && (roles.includes(Role.ADMIN) || roles.includes(Role.ORG_ADMIN) || roles.includes(Role.OPPORTUNITY_PROVIDER))) {
```
[the remark this answers is not in the document]
Comment (farjanacan) [reply components/Story/AddStory.js]: I've removed the role 'opportunity provider'.

File: components/Story/EditableStory.js [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+import { useDispatch } from 'react-redux'
 import { StoryDetail } from '../../components/Story/StoryDetail'
 
 export const stories = []
 
-export const EditableStory = ({ story }) => {
+export const EditableStory = ({ roles, story }) => {
   const [editing, setEditing] = useState(false)
-  const [storyview, setStoryview] = useState(story)
+  const dispatch = useDispatch()
```
[the remark this answers is not in the document]
Comment (farjanacan) [reply components/Story/EditableStory.js]: I'm getting the following error if I don't use the useDispatch function and instead pass the dispatch to the prop list. 
error
code

File: components/Story/StoryForm.js [hunk 4]
```diff
@@ -31,12 +31,14 @@ export const StoryForm = ({ story, onSubmit }) => {
...
-            <input value={storyed.name} onChange={(e) => setStoryed({ name: e.target.value, body: storyed.body })} />
+            <input value={storyed.name} onChange={(e) => setStoryed({ ...storyed, name: e.target.value })} />
           </Form.Item>
           <Form.Item label={storyDescription}>
-            <TextArea cols='35' value={storyed.body} onChange={(e) => setStoryed({ name: storyed.name, body: e.target.value })} />
+            <TextArea cols='35' value={storyed.body} onChange={(e) => setStoryed({ ...storyed, body: e.target.value })} />
           </Form.Item>
-          <Button type='primary' onClick={() => onSubmit(storyed)}>Save</Button>
+          <Button type='primary' onClick={() => onSubmit(storyed)}>
+            <FormattedMessage id='story.publish' defaultMessage='Publish' description='Button to publish a story on updates tab' />
```
[the remark this answers is not in the document]
Comment (farjanacan) [reply components/Story/StoryForm.js]: I'm still working on 'save as draft' feature, that's why the button is not there yet.

File: server/api/story/__tests__/story.spec.js (hunk 1 above)
Comment (farjanacan) [reply server/api/story/__tests__/story.spec.js]: I got 1 result after merging 'beforeEach' with 'before'. 
I've added two stories with the same parent to the fixture and checked the got.parent result. The tests passed.

File: components/Story/AddStory.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (avowkind) [reply components/Story/AddStory.js]: Thanks - but you still need a bit more. 
The requestor of the op is the person who is going to be adding updates.   e.g. me === op.requestor  but the org admin can also update stories if they are orgAdmin for the same org that is referenced in the offerOrg field of the op. 
If session.me.role.includes(ORG_ADMIN) then there will also be session.me.orgAdminFor [] listing the ids of the orgs for which the person is an admin.  so you should check
 session.me.role.includes(ORG_ADMIN) && session.me.orgAdminFor.includes(op.offerOrg)

These checks should take place earlier in the panel page. collect them together into a canAddNew flag and then only show the button if its true e.g.
canAddNew && <AddStory onSubmit={handler} />   
This allows you to remove all the checking from AddStory itself.

File: components/Story/EditableStory.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (avowkind) [reply components/Story/EditableStory.js]: Ok stick with useDispatch

**Pull request merged**